pkg/api/v20170701/types.go

```diff
@@ -98,19 +101,19 @@ const (
 
 // OrchestratorProfile contains Orchestrator properties
 type OrchestratorProfile struct {
-	OrchestratorType    OrchestratorType    `json:"orchestratorType"`
+	OrchestratorType    OrchestratorType    `json:"orchestratorType" validate:"required"`
 	OrchestratorVersion OrchestratorVersion `json:"orchestratorVersion"`
 }
 
 // MasterProfile represents the definition of master cluster
 type MasterProfile struct {
-	Count                    int    `json:"count"`
-	DNSPrefix                string `json:"dnsPrefix"`
-	VMSize                   string `json:"vmSize"`
-	OSDiskSizeGB             int    `json:"osDiskSizeGB,omitempty"`
+	Count                    int    `json:"count" validate:"required,eq=1|eq=3|eq=5"`
```

Comment:
If we specify `count` with tags like this. It will be automatically validated. No need to write functions to handle it. TAL this file, and compare with what i have removed in the validate.go file. You will get the idea here.